Applications/WebPortal/WebApp/library/source/components/country_selection_box.tsx

```diff
@@ -10,11 +12,126 @@ interface Properties {
   value: Nexus.CountryCode;
 
   /** Whether the selection box is read only. */
-  readonly: boolean;
+  readonly?: boolean;
 
   /** The event handler called when the selection changes. */
   onChange?: (currency: Nexus.CountryCode) => void;
+
+  displaySize: DisplaySize;
+}
+
+interface State {
+  options: any[];
```

Comment:
Only state that triggers a ```render``` should be included in the ```State``` interface. In this case you never update or trigger a render based on ```options```, so make ```options``` a private field rather than a part of ```State```.